### PR TITLE
Fix compatibility with pulpcore 3.16

### PR DESCRIPTION
**Pull request by goosemania**
[noissue]

### CODE REVIEW

**Comment by goosemania** [comment]
@dralley , @dkliban , @ipanova , CI is green but the code in `content.py` is ugly.
I would prefer to move large pieces into separate functions and just call them wrapped in `sync_to_async`, but the current code is a mixture of sync and async stuff, so it seems to be hard to do that without major rewrite/refactor.
Thoughts or suggestions?

**Review by dkliban — APPROVED** [review]

**Comment by dralley** [comment]
>I would prefer to move large pieces into separate functions and just call them wrapped in sync_to_async, but the current code is a mixture of sync and async stuff, so it seems to be hard to do that without major rewrite/refactor.

Yeah, I don't think we should spend any more time on it than we absolutely need to.  I'm fine with not doing any substantial refactors.

File: pulp_2to3_migration/app/plugin/content.py [hunk 1]
```diff
@@ -246,16 +249,21 @@ def get_remote_by_importer_id(importer_id):
                 select_extra.append('pulp2content__pulp2_repo')
 
             pulp_2to3_detail_qs = pulp_2to3_detail_qs.select_related(*select_extra)
-            for pulp_2to3_detail_content in pulp_2to3_detail_qs.iterator(chunk_size=800):
+            async for pulp_2to3_detail_content in sync_to_async_iterable(
+                    pulp_2to3_detail_qs.iterator(chunk_size=800)
+            ):
                 dc = None
                 pulp2content = pulp_2to3_detail_content.pulp2content
+                await sync_to_async(pulp2content.refresh_from_db)()  # force model instance to be
```
Comment (dralley) [inline pulp_2to3_migration/app/plugin/content.py:257]: What's going on here?

Comment (goosemania) [reply pulp_2to3_migration/app/plugin/content.py:257]: pulp2content is used in many places later, this is to force to load fully from the db without a need to do db requests later when different fields are accessed.

**Comment by ipanova** [comment]
> @dralley , @dkliban , @ipanova , CI is green but the code in `content.py` is ugly. I would prefer to move large pieces into separate functions and just call them wrapped in `sync_to_async`, but the current code is a mixture of sync and async stuff, so it seems to be hard to do that without major rewrite/refactor. Thoughts or suggestions?

I'd leave this is as is. The changes are easily readable.

File: pulp_2to3_migration/app/plugin/content.py (hunk 1 above)
Comment (ipanova) [reply pulp_2to3_migration/app/plugin/content.py:257]: I think whether get() or refresh_from_db() is used there is not much of the difference. refresh_from_db() calls anyway into get() its just does not load deferred fields and clears cached relations https://github.com/django/django/blob/main/django/db/models/base.py#L652

**Review by ipanova — APPROVED** [review]

File: pulp_2to3_migration/app/plugin/content.py [hunk 2]
```diff
@@ -246,16 +250,22 @@ def get_remote_by_importer_id(importer_id):
...
 
             pulp_2to3_detail_qs = pulp_2to3_detail_qs.select_related(*select_extra)
-            for pulp_2to3_detail_content in pulp_2to3_detail_qs.iterator(chunk_size=800):
+            async for pulp_2to3_detail_content in sync_to_async_iterable(
+                    pulp_2to3_detail_qs.iterator(chunk_size=800)
+            ):
                 dc = None
-                pulp2content = pulp_2to3_detail_content.pulp2content
+                pulp2content = await sync_to_async(Pulp2Content.objects.get)(
+                    pulp_2to3_detail_content.pulp2content_id
```
Comment (dralley) [inline pulp_2to3_migration/app/plugin/content.py:258]: I think it needs `pk=`, my bad

Comment (goosemania) [reply pulp_2to3_migration/app/plugin/content.py]: yeah, I just got to it and figured that, we are perfectly in sync :)

**Pull request merged**